fortuna/src/command/register_provider.rs

```diff
@@ -59,7 +59,7 @@ pub async fn register_provider(opts: &RegisterProviderOptions) -> Result<()> {
             commitment,
             bincode::serialize(&commitment_metadata)?.into(),
             commitment_length,
-            bincode::serialize(&opts.uri)?.into(),
+            bincode::serialize(&chain_config.provider_uri)?.into(),
```

Comment:
This seems wrong to me. The chain config as it stands is a factory that lets you instantiate an Entropy contract on a blockchain. I think you want all of the provider information being registered to be explicitly passed in to this function, not implied from the chain config.